filebeat/input/unix/input.go

```diff
@@ -18,118 +18,97 @@
 package unix
 
 import (
+	"bufio"
 	"fmt"
-	"sync"
+	"net"
 	"time"
 
-	"github.com/elastic/beats/v7/filebeat/channel"
-	"github.com/elastic/beats/v7/filebeat/harvester"
-	"github.com/elastic/beats/v7/filebeat/input"
+	input "github.com/elastic/beats/v7/filebeat/input/v2"
+	stateless "github.com/elastic/beats/v7/filebeat/input/v2/input-stateless"
 	"github.com/elastic/beats/v7/filebeat/inputsource"
 	netcommon "github.com/elastic/beats/v7/filebeat/inputsource/common"
 	"github.com/elastic/beats/v7/filebeat/inputsource/unix"
 	"github.com/elastic/beats/v7/libbeat/beat"
 	"github.com/elastic/beats/v7/libbeat/common"
-	"github.com/elastic/beats/v7/libbeat/common/cfgwarn"
-	"github.com/elastic/beats/v7/libbeat/logp"
+	"github.com/elastic/beats/v7/libbeat/feature"
+	"github.com/elastic/go-concert/ctxtool"
 )
 
-func init() {
-	err := input.Register("unix", NewInput)
-	if err != nil {
-		panic(err)
-	}
-}
-
-// Input for Unix socket connection
-type Input struct {
-	mutex   sync.Mutex
-	server  *unix.Server
-	started bool
-	outlet  channel.Outleter
-	config  *config
-	log     *logp.Logger
+type server struct {
+	config
+	splitFunc bufio.SplitFunc
 }
 
-// NewInput creates a new Unix socket input
-func NewInput(
-	cfg *common.Config,
-	connector channel.Connector,
-	context input.Context,
-) (input.Input, error) {
-	cfgwarn.Beta("Unix socket support is beta.")
-
-	out, err := connector.Connect(cfg)
-	if err != nil {
-		return nil, err
+func Plugin() input.Plugin {
+	return input.Plugin{
+		Name:       "unix",
+		Stability:  feature.Beta,
+		Deprecated: false,
+		Info:       "unix socket server",
+		Manager:    stateless.NewInputManager(configure),
 	}
+}
 
-	forwarder := harvester.NewForwarder(out)
-
-	config := defaultConfig
-	err = cfg.Unpack(&config)
-	if err != nil {
+func configure(cfg *common.Config) (stateless.Input, error) {
+	config := defaultConfig()
+	if err := cfg.Unpack(&config); err != nil {
 		return nil, err
 	}
 
-	cb := func(data []byte, metadata inputsource.NetworkMetadata) {
-		forwarder.Send(beat.Event{
-			Timestamp: time.Now(),
-			Fields: common.MapStr{
-				"message": string(data),
-			},
-		})
-	}
+	return newServer(config)
+}
 
+func newServer(config config) (*server, error) {
```

Comment:
the constructor (in this case `newServer`) should not depend on `common.Config`. The `configure` function is introduced to unpack the config and pass it to the typed constructor.